Tests/TestUtils/XCTestCase+AnyCodableAsserts.swift

```diff
@@ -488,21 +488,34 @@ extension XCTestCase {
                 wildcardIndexes = Set(0..<expected.count)
                 expectedIndexes.removeAll()
             } else {
-                // TODO: update this to be flat? since there's only 1 operation now instead of 3
-                // Strongly validates index notation: "[123]"
+                // Finds wildcard indexes and extract their index values
+                
+                // Strongly validates array style access format: "[123]"
+                // Then filters all the path end keys that have valid array style access format, and returns only the string between the brackets
+                // Ex: "[123]" -> "123", "[ab12]" -> "ab12"
+                
+                // Wildcard index processing:
+                // 1. Filters the index values to the ones that have the wildcard marker `"*"`
+                // 2. Removes all asterisk characters from the index string
+                // 3. Casts the end result to an Int, discarding index if it is not a valid Int (sanitizes invalid strings)
+                // 4. Creates a unique set of the final values that pass all the above steps
+                
+                // Discard wildcard indexes that are out of bounds of the available expected indexes
+                // Then remove all wildcard indexes from the valid expected indexes, so validation is not performed twice
+                
                 let arrayIndexValueRegex = #"^\[(.*?)\]$"#
                 let indexValues = Set(pathEndKeys
                     .flatMap { getCapturedRegexGroups(text: $0, regexPattern: arrayIndexValueRegex) })
-
-                func filterConvertAndIntersect(condition: (String) -> Bool, replacement: @escaping (String) -> String = { $0 }) -> Set<Int> {
-                    var result = Set(indexValues.filter(condition).compactMap { Int(replacement($0)) })
-                    let intersection = expectedIndexes.intersection(result)
-                    result = intersection
-                    expectedIndexes.subtract(intersection)
-                    return result
-                }
-
-                wildcardIndexes = filterConvertAndIntersect(condition: { $0.contains("*") }, replacement: { $0.replacingOccurrences(of: "*", with: "") })
+                
+                
+                var result = Set(indexValues
+                    .filter { $0.contains("*") }
+                    .compactMap { Int($0.replacingOccurrences(of: "*", with: "")) })
+                
+                let intersection = expectedIndexes.intersection(result)
```

Comment:
nit: Update variable names with more context and add comments to make the code easy to read. Something on these lines:

```
// Filters all the entries with wildcard character `*` and returns the index marked with wildcard.
var listOfIndexes = Set(indexValues
                    .filter { $0.contains("*") }
                    .compactMap { Int($0.replacingOccurrences(of: "*", with: "")) })
```

```
// result of regex matching with list of wildcard indexes
let wildcardIndexes = expectedIndexes.intersection(result)
````